pkg/detector/detector.go

```diff
@@ -700,15 +701,19 @@ func (d *ResourceDetector) GetUnstructuredObject(objectKey keys.ClusterWideKey)
 // ClaimPolicyForObject set policy identifier which the object associated with.
 func (d *ResourceDetector) ClaimPolicyForObject(object *unstructured.Unstructured, policy *policyv1alpha1.PropagationPolicy) (string, error) {
 	policyID := policy.Labels[policyv1alpha1.PropagationPolicyPermanentIDLabel]
-
+	generation := strconv.FormatInt(policy.Generation, 10)
 	objLabels := object.GetLabels()
 	if objLabels == nil {
 		objLabels = make(map[string]string)
 	} else if len(objLabels) > 0 {
 		// object has been claimed, don't need to claim again
 		if !excludeClusterPolicy(objLabels) &&
 			objLabels[policyv1alpha1.PropagationPolicyPermanentIDLabel] == policyID {
-			return policyID, nil
+			if !util.IsLazyActivationEnabled(policy.Spec.ActivationPreference) {
+				return policyID, nil
+			} else if object.GetAnnotations()[policyv1alpha1.PropagationPolicyGenerationAnnotation] == generation {
+				return policyID, nil
+			}
```

Comment:
I don't quite understand the meaning of these lines of code. Can you add some comments to explain these lines?

Reply:
@chaosi-zju please take a look again.